File: app/src/main/java/com/example/haushaltsapp/types/SupplyEntry.java
package com.example.haushaltsapp.types;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.time.ZonedDateTime;

class SupplyEntry {

    @NonNull
    private String name;

    @NonNull
    private String amount;

    @Nullable
    private ZonedDateTime expiry;

    public SupplyEntry(@NonNull String name, @NonNull String amount, @Nullable ZonedDateTime expiry) {
        this.name = name;
        this.amount = amount;
        this.expiry = expiry;
    }

    @NonNull
    public String getName() {
        return name;
    }

    public void setName(@NonNull String name) {
        this.name = name;
    }

    @NonNull
    public String getAmount() {
        return amount;
    }

    public void setAmount(@NonNull String amount) {
        this.amount = amount;
    }

    @Nullable
    public ZonedDateTime getExpiry() {
        return expiry;
    }

    public void setExpiry(@Nullable ZonedDateTime expiry) {
        this.expiry = expiry;
    }
}

File: app/src/main/java/com/example/haushaltsapp/types/SupplyEntry.kt
package com.example.haushaltsapp.types

import java.time.ZonedDateTime

data class SupplyEntry(
        //val id: Int,
        var id: String,
        var name: String,
        var amount: String,
        var expiry: ZonedDateTime?
) {
    @Suppress("unused")
    constructor() : this("", "", "", null)
    constructor(amount: String) : this() {

    }

    constructor(name: String, amount: String) : this("", "", "", null) {

    }



    fun toSummary(): SupplySummary = SupplySummary(id, name)
}